algorithmic_efficiency/workloads/criteo1tb/workload.py

```diff
@@ -85,9 +88,14 @@ def build_input_queue(self,
         vocab_sizes=self.vocab_sizes,
         num_batches=num_batches,
         repeat_final_dataset=repeat_final_dataset)
-    for batch in iter(ds):
-      batch = jax.tree_map(lambda x: x._numpy(), batch)  # pylint: disable=protected-access
-      yield batch
+
+    # Separate function is necessary because the code above has to be executed
+    # when build_input_queue is called (not when next() is first called on it).
+    def _input_queue_generator():
```

Comment:
In this case no separate function is necessary. This was only needed for WMT because the tokenizer had to be created in all processes.